### PR TITLE
piaf.0.1.0 requires at least bigstringaf.0.8.0 and stdlib-shims

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

**Comment by mseri** [comment]
See: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/db7fa6ffdfb646f88a35c9457b000bbbe18b05bb/variant/opam-2.1,compilers,4.08,piaf.0.1.0,lower-bounds

```
# Error: Conflict between the following libraries:
# - "piaf.httpaf" in _build/default/vendor/httpaf/lib
# - "httpaf" in /home/opam/.opam/4.08/lib/httpaf
#   -> required by library "gluten" in /home/opam/.opam/4.08/lib/gluten
#   -> required by library "piaf.h2-lwt" in _build/default/vendor/ocaml-h2/lwt
# Error: Conflict between the following libraries:
# - "piaf.httpaf" in _build/default/vendor/httpaf/lib
# - "httpaf" in /home/opam/.opam/4.08/lib/httpaf
#   -> required by library "gluten" in /home/opam/.opam/4.08/lib/gluten
# Error: Conflict between the following libraries:
# - "piaf.httpaf" in _build/default/vendor/httpaf/lib
# - "httpaf" in /home/opam/.opam/4.08/lib/httpaf
#   -> required by library "gluten" in /home/opam/.opam/4.08/lib/gluten
#   -> required by library "piaf.httpaf-lwt" in
#      _build/default/vendor/httpaf/lwt
# Error: Conflict between the following libraries:
# - "piaf.httpaf" in _build/default/vendor/httpaf/lib
# - "httpaf" in /home/opam/.opam/4.08/lib/httpaf
#   -> required by library "gluten" in /home/opam/.opam/4.08/lib/gluten
#   -> required by library "piaf.h2-lwt" in _build/default/vendor/ocaml-h2/lwt
#   -> required by library "piaf.h2-lwt-unix" in
#      _build/default/vendor/ocaml-h2/lwt-unix
```

I think you need to add a conflict with httpaf since you are vendoring it

**Comment by dinosaure** [comment]
> I think you need to add a conflict with httpaf since you are vendoring it

/cc @anmonteiro but yeah, then I can not really figure out what this conflict can imply then for the rest of packages.

**Comment by mseri** [comment]
Yeah but with vendoring that is unavoidable, unless you rewrite the vendored dependencies to prefix the modules with something

File: packages/piaf/piaf.0.1.0/opam [hunk 1]
```diff
@@ -29,6 +30,7 @@ depends: [
   "alcotest" {with-test}
   "alcotest-lwt" {with-test}
 ]
+conflicts: [ "httpaf" ]
```
Comment (mseri) [inline packages/piaf/piaf.0.1.0/opam:33]:
```suggestion
conflicts: [
  "httpaf"
  "result" {< "1.5"}
]
```

File: packages/piaf/piaf.0.1.0/opam [hunk 2]
```diff
@@ -13,8 +13,9 @@ depends: [
   "ocaml" {>= "4.08"}
   "dune" {>= "2.5"}
   "angstrom"
-  "faraday"
-  "bigstringaf" {>= "0.5.0"}
+  "faraday" {>= "0.8.1"}
+  "bigstringaf" {>= "0.8.0"}
+  "stdlib-shims"
   "logs"
   "lwt"
```
Comment (mseri) [inline packages/piaf/piaf.0.1.0/opam:20]:
```suggestion
  "lwt" {>= "5.3.0"}
```
Uses `Lwt_result.Syntax`

File: packages/piaf/piaf.0.1.0/opam (hunk 1 above)
Comment (mseri) [reply packages/piaf/piaf.0.1.0/opam:33]: Due to `Result.map` not found, installing `result 1.0` in the lower bounds

**Comment by mseri** [comment]
Thanks

**Pull request merged**